build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/release/SetCompatibleVersionsTask.java

```diff
@@ -28,28 +29,44 @@
 
 public class SetCompatibleVersionsTask extends AbstractVersionsTask {
 
+    private Version thisVersion;
+    private Version releaseVersion;
     private Map<String, Integer> versionIds = Map.of();
 
     @Inject
     public SetCompatibleVersionsTask(BuildLayout layout) {
         super(layout);
     }
 
+    public void setThisVersion(Version version) {
+        thisVersion = version;
+    }
+
     @Option(option = "version-id", description = "Version id used for the release. Of the form <VersionType>:<id>.")
     public void versionIds(List<String> version) {
         this.versionIds = splitVersionIds(version);
     }
 
+    @Option(option = "release", description = "The version being released")
+    public void releaseVersion(String version) {
+        releaseVersion = Version.fromString(version);
+    }
+
     @TaskAction
     public void executeTask() throws IOException {
         if (versionIds.isEmpty()) {
             throw new IllegalArgumentException("No version ids specified");
         }
+
+        if (releaseVersion.getMajor() < thisVersion.getMajor()) {
```

Comment:
Should we put this conditional above so that it doesn't matter if we pass empty IDs since it doesn't matter in this case?

Reply:
Every release should have version ids associated with it, so there should always be something in the map